runtimes/common/src/bonded_currencies/mod.rs

```diff
@@ -0,0 +1,291 @@
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::mint_into(asset, who, amount)
+	}
+	fn burn_from(
+		asset: Self::AssetId,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+		force: Fortitude,
+	) -> Result<Self::Balance, DispatchError> {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::burn_from(asset, who, amount, precision, force)
+	}
+	fn shelve(asset: Self::AssetId, who: &AccountId, amount: Self::Balance) -> Result<Self::Balance, DispatchError> {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::shelve(asset, who, amount)
+	}
+	fn restore(asset: Self::AssetId, who: &AccountId, amount: Self::Balance) -> Result<Self::Balance, DispatchError> {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::restore(asset, who, amount)
+	}
+	fn transfer(
+		asset: Self::AssetId,
+		source: &AccountId,
+		dest: &AccountId,
+		amount: Self::Balance,
+		preservation: Preservation,
+	) -> Result<Self::Balance, DispatchError> {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::transfer(asset, source, dest, amount, preservation)
+	}
+
+	fn set_balance(asset: Self::AssetId, who: &AccountId, amount: Self::Balance) -> Self::Balance {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::set_balance(asset, who, amount)
+	}
+}
+
+impl<
+		Left: fungible::Inspect<AccountId>
+			+ AccountTouch<(), AccountId, Balance = <Left as fungible::Inspect<AccountId>>::Balance>,
+		Right: fungibles::Inspect<AccountId>
+			+ AccountTouch<Right::AssetId, AccountId, Balance = <Left as fungible::Inspect<AccountId>>::Balance>,
+		Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+		AssetKind: AssetIdTraits,
+		AccountId,
+	> AccountTouch<AssetKind, AccountId> for NativeAndForeignAssets<Left, Right, Criterion, AssetKind, AccountId>
+{
+	type Balance = <Left as fungible::Inspect<AccountId>>::Balance;
+
+	fn deposit_required(asset: AssetKind) -> Self::Balance {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::deposit_required(asset)
+	}
+
+	fn should_touch(asset: AssetKind, who: &AccountId) -> bool {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::should_touch(asset, who)
+	}
+
+	fn touch(asset: AssetKind, who: &AccountId, depositor: &AccountId) -> DispatchResult {
+		UnionOf::<Left, Right, Criterion, AssetKind, AccountId>::touch(asset, who, depositor)
+	}
+}
+
+impl<
+		Left: fungible::Inspect<AccountId>,
+		Right: Inspect<AccountId, Balance = <Left as fungible::Inspect<AccountId>>::Balance>,
+		Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+		AssetKind: AssetIdTraits,
+		AccountId,
+	> Inspect<AccountId> for NativeAndForeignAssets<Left, Right, Criterion, AssetKind, AccountId>
+{
+	fn decimals(asset: Self::AssetId) -> u8 {
+		match Criterion::convert(asset) {
+			Left(()) => 15u8,
+			Right(a) => Right::decimals(a),
+		}
+	}
+
+	fn name(asset: Self::AssetId) -> Vec<u8> {
+		match Criterion::convert(asset) {
+			Left(()) => Vec::new(),
```

Comment:
same for name and symbol; if we were to expose pool info via a runtime call this would be really helpful for example.

Reply:
I hard coded KILT in the runtime api pr: https://github.com/KILTprotocol/kilt-node/pull/828. 

I hope it's fine to hard code those, otherwise I have to introduce another generic, to fetch the right symbol and  name, as they are different in the different runtimes.

Reply:
I changed my mind. I have already introduced the metadata in this pr. Please check out this commit: [0dcf5b3](https://github.com/KILTprotocol/kilt-node/pull/818/commits/0dcf5b38aaf642cb694d3e8cc7cb222218a091ea)